README.md
# dotfiles

summarize my dotfiles.

## Require

* ansible

## Usage

    % ansible-playbook -i <host> <thing>.yml

or playbook locally,

    % ansible-playbook -i localhost, -c local <thing>.yml

## License

Copyright 2022 &copy; Hiroshi Kimura




CHANGELOG.md
# CHANGELOG

## Unreleased
- 複数の yml を一気に実行する。Makefile ではない。
- clojure/installer - 時代遅れになるインストーラを直置きするより、
  インストーラの見つかる URL を置く。
- emacs は .cofig/emacs/ に動かすと cider がすんなり動かない。


## 1.4.2 - 2022-12-29
### Added
- dev.weavejester/medley to clojure/versions/deps.edn

## 1.4.1 - 2022-12-26
### Added
- empty emacs.d/inits-diabled/.placeholder for git.

## 1.4.0 - 2022-12-26
### Added
- clojure/versions

### Changed
- emacs.yml: check existence of .emacs.d, .emacs.d/inits and .emacs.d.inits-disabled

## 1.4.0-SNAPSHOT
- git を `XDG_CONFIG_HOME/git` へ。

## 1.3.1 - 2022-10-01
### Added
- brew/pandoc
  pandoc は前から使っていたが？

### removed
- brew/jupyterlab

## 1.3.0 - 2022-09-18
### Added
- bash-git-prompt
- zsh/fix-opt-zshrc.sh
  brew の zsh-git-prompt はいまだに python をポイントしている。
  python3 にスクリプトを書き換えないとプロンプトがリポジトリの状態を
  反映しない。


## 1.2.13.1 - 2022-09-15
- Your GitHub access token has been found in hkim0331/dotfiles and revoked
  zshenv をgit 下で管理するのをやめる。zshenv_local 作戦もやめ。

## 1.2.13 - 2022-09-14
- zsh.yml overwrite existing ~/.zshenv.
  instead, read ~/.zshenv_local if exists

## 1.2.12 - 2022-09-09
- some nrepl aliases from practicalli

## 1.2.11 - 2022-09-08
- opecv はコメントアウトしない。brew bundle cleanup で消えてしまう。
  clojure からの呼び出しのために、reinstall --build-from-source が必要になる。
- git/autocrlf=true は Windows オンリー。マックでそれやるとおかしなことになる。
- common-lisp/roswell
  いったん、brew uninstall roswell してから brew install roswell し直し。
  これで imac3 の common-lisp は復活した。
- alias ec for emacsclient

## 1.2.10 - 2022-09-05
### hotfix
- .calva, .clj-kondo, .cpcacpe, .lsp はプロジェクトのトップ以外にも
 作られる。1.2.9 ではまずい。たとえば、.gitignore に次を書いていても、
```
.clj-kondo/.cache/
```
`sort/.clj-kondo/.cache/v1/...`  にはマッチしない。

## 1.2.9 - 2022-09-05
### Changed
- git/gitignore_global
  プロジェクトのサブディレクトリに .cpcache, .clj-kondo, .lsp が
  作成されることがある。ignore するのは .cache だけで OK か?

## 1.2.8 - 2022-09-04
### Fix
- emacs.yml inits/ 下のファイルはコピーじゃいけない。sync か。

## 1.2.7 - 2022-09-04
### Fixed
- .cljstyle のインストールを間違っていた。

### Added
- emacs/inits/60_roswell.el

### Changed
- emacs/inits order of loading packages

### Added
- rc/myclirc

## 1.2.6 - 2022-08-30
- updated ~/.cljstyle by `cljstyle migrate`
- updated grope to exclude .shadow-cljs folder

## 1.2.5
- `postgresql@14` instead of `postgresql`.

## 1.2.4
- remove mysql-client
  could not make mariadb dump using mysql-client, `mysqldump`.
  moved to mariadb/mycli.

## 1.2.3 - 2022-08-28
* `git vt` is aliased "tag -l -n --sort=-v:refname"
* `git/deps.edn` is a manual copy of
  `/opt/homebrew/Cellar/clojure/1.11.1.1155/deps.edn` (must improve)
* improved alias `grope` to exclude some meaningless subdirectories.

## 1.2.2 - 2022-08-23
* came back `clojure/cljstyle` for 1 line between defns.
  default is 2 lines.

## 1.2.1 - 2022-08-22
* fixed typo `trash-put`
* learn how to use cljstyle. also removed ~/.cljstyle

## 1.2.0 - 2022-08-22
### Changed
* Brewfile force: no
### Added
* brew/mycli
* brew/trash-cli

## 1.1.0 - 2022-08-21
* vim
* clojure
* rc

## 1.0.2 - 2002-08-21

## 1.0.1 - 2022-08-20
* added Makefile

## 1.0.0 - 2022-08-20
出直し。

* Brewfile
*


## 0.12.3 - 2022-04-01
- 混乱をまとめる。python is python3, pip is pip3.

## 0.12.2 - 2022-04-01
- `python` is an alias of `python3`

## 0.12.0 - 2022-02-14
### Added
- new zsh alias, psj and psq
### Changed
- unignore *.sql
### Fixed
* newer straight does not have (require 'a) line. was it a bug?
* cider loading
  (require 'cl-lib)
  (require 'subr-x)
  ;; hkimura, 2021-09-28
  ;;(require 'a)
  (require 'parseclj-lex)

## 0.11.0-SNAPSHOT
### Removed
- utils/ and utils.yml for being created `utils` project.
### Fixme
- ChromeOS
### Changed
- gitignore_global do not ignore *.sql - 2021-10-04
- ubuntsu-clojure.yml 2021-09-26 linux-install-1.10.3.986.sh
- utils/trash.clj: polish up code.
- zsh/zprofile: include coreutils/libexec/gnubin into PATH.
### Added
- lxd/bin/lxc-default: make lxc guest's interface to default (lxdbr0).
- emacs.d/inits/37_inf-clojure.el


## 0.10.0 - 2021-09-08
### Changed
- utils/wol.clj: wol [on|off|stat] host1 host2 ...
  parallel execution with pmap.

## 0.9.43-SNAPSHOT
### Added
- inventory/bionic.local -- nuc7 lxd guest

### Removed
- removed prompt git branch section from bash/bashrc

## 0.9.42 - 2021-06-19
- utils/symlink-clj: 現在ディレクトリ中の *.clj を
  拡張子なしファイル名からシンボリックリンク。
- stop using apt-cacher-ng


## 0.9.42-SNAPSHOT - 2021-06-03
- favicon: remove work files after making favicon.ico.


## 0.9.41 - 2021-05-27
### Changed
- rename emacs as emacs.d
- aspell-dictionary "en_US" in emacs.d/init.el
- aspell def in emacs.d/inits/90_my.el

## 0.9.40 - 2021-05-21
BUG: 0.9.39 does not work as expected.
### Changed
- rename mkss as cprl.
### Added
- outdated.clj from babashka site.


## 0.9.39 - 2021-05-20
### Changed
* utils.yml: ファイル `*.clj` は特別に `dest: ~/bin/*` でインストール。
  canceled. see 0.9.40.

## 0.9.38 - 2021-05-20
### Added
- rename all-but-me.clj to into.clj.
- utils/all-but-me.clj: (consider command name)
- utils/mkss.clj: make substanciate symbolic links.
### Changed
- emacs/inits/90_my.el: dired-omit-mode


## 0.9.37 - 2021-05-15
### Added
- utils/base64.clj: base64 encoder/decoder.
- utils/favicon.clj: favicon maker.
- utils/trash.clj, update utils.yml.


## 0.9.36 - 2021-05-13
### Changed
want opencv java bindings.
- brew uninstall openjdk && brew install --cask zulu
- zsh/zprofile: define JAVA_HOME and CPPFLAGS for zulu.
  PATH included $JAVA_HOME/bin.

## 0.9.35 - 2021-05-11
### Added
- utils/rmlss for removing all local snap shots.
### Fixed
- inventory/imac3.local. wrongly defined as `localhost`.
### Changed
- chmod 0644 utils/*.

## 0.9.34 - 2021-05-08
### Added
- utils/bak backup files with today's timestamp, -yyyy-mm-dd.

## 0.9.33 - 2021-05-02
### Added
- alias `llss` for list local snapshots.
- rewrote shell script `utils/wol` with babashka, as `wol.clj`.
- alias `tm` for launch TimeMachine.

### 0.9.30 - 2021-03-27
* removed useless files.

### 0.9.28 - 2021-03-27
* added utils/p asciidoc previewer.

### 0.9.27 - 2021-02-25
* changed lxd/bin/lxc-prep.

emacs/inits/90_my 2.el
;;; hkim prefers.
;;; custom?
(unless window-system (menu-bar-mode -1))
(setq global-hl-line-mode nil)
(setq vc-follow-symlinks t)
(setq mouse-drag-copy-region t) ;; require scrolling?
(setq-default tab-width 2 indent-tabs-mode nil)
(global-set-key "\C-cg" 'goto-line) ; M-g M-g
(global-set-key "\C-c\C-m" 'browse-url-at-point)
(add-hook 'after-save-hook
          'executable-make-buffer-file-executable-if-script-p)

;;; aspell
(when (executable-find "/usr/local/bin/aspell")
  (setq-default ispell-program-name "/usr/local/bin/aspell"))
(when (executable-find "/opt/homebrew/bin/aspell")
  (setq-default ispell-program-name "/opt/homebrew/bin/aspell"))
(with-eval-after-load "ispell"
  '(add-to-list 'ispell-skip-region-alist '("[^\000-\377]+")))

;;; dired-x, if gls found, use it.
;;; C-x M-o to toggle omit state.
;;; default omit-mode 2021-05-15.
(require 'dired-x)
(setq dired-omit-files (concat dired-omit-files "\\|^\\..+$"))
(when (executable-find "/usr/local/bin/gls")
  (setq insert-directory-program "/usr/local/bin/gls"))
(when (executable-find "/opt/homebrew/bin/gls")
  (setq insert-directory-program "/opt/homebrew/bin/gls"))
;; prefer omit-mode, enable below.
;; no effect.
;;(setq-default dired-omit-files-p t)
;; this is OK.
;;(add-hook 'dired-mode-hook 'dired-omit-mode)

(require 'wdired)
(define-key dired-mode-map "r" 'wdired-change-to-wdired-mode)

;;; whitespace
(require 'whitespace)
(global-whitespace-mode 1) ; デフォルトで視覚化を有効にする。
(set-face-background 'whitespace-space "red")
(setq whitespace-space-regexp "\\(\u3000\\)")
(setq whitespace-style '(face trailing tabs spaces))

;;; private defuns
(defun eshell/clear ()
  "Clear eshell buffer."
  (interactive)
  (let ((inhibit-read-only t))
    (erase-buffer)))

(defun now()
  "Insert timestamp ISO format."
  (interactive)
  (insert (format-time-string "%Y-%m-%d %H:%M:%S" (current-time))))

(defun today()
  "Insert date string with ISO format."
  (interactive)
  (insert (format-time-string "%Y-%m-%d" (current-time))))



;; CHANGED: was defined as cl-label. 2021-02-13.
(defun first-file-exists (files)
  (cond
   ((null files) nil)
   ((file-exists-p (car files)) (car files))
   (t (first-file-exists (cdr files)))))

(defun memo ()
  "Open private memo.txt."
  (interactive)
  (find-file
   (first-file-exists '("~/.emacs.d/memo.txt"
                        "~/Documents/memo.txt"
                        "~/projects/dotfiles/memo.txt"))))

;;; roswell
;;; Use 'ros emacs' instead. (hkimura 2020-09-14)

;;; super-save
;;; https://github.com/bbatsov/super-save
;;; look 30_super-save.el.
;; (super-save-mode +1)
;; (setq super-save-auto-save-when-idle t
;;       super-save-idle-duration 2)

;;; my profile
(setenv "LANG" "ja_JP.UTF-8")
(setq user-full-name "Hiroshi Kimura")
(setq user-mail-address "hiroshi.kimura.0331@gmail.com")

